Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -412,6 +412,64 @@ and/or costly.
+    safety requirements rely on multiple subsystems, possibly across different
+    software layers.
+    
+    As usual, it is extremely important to assess upfront the requirements
+    for availability.
+
+    A requirement for simply detecting the potential interference can be easier
+    to deal with, than a requirement to ensure continued safety operations,
+    in the presence of interference.
+
+    Leaving aside, for now, considerations about the latter scenario, the former
+    one still carries a need for the detection to happen within a maximum interval,
+    measured from the beginning of the interference.
+
+    With these two requirements in mind:
+    * detecting interference
+    * detecting the interference within a maximum allowed time
+      
+    One can attempt to isolate those subsystems directly involved, and investigate
+    the possibility of periodically validating its continued integrity.
+    
+    Here, the basic concept is that one could be able to provide - periodically -
+    a preset stimuli, and compare the live output of the system with the
+    pre-calculated expected one, that the system should produce, when free from
+    interference.
+    
+    Taking as example an image recognition pipeline, tasked with receiving an input
+    from a sensor and recognising items from a given set (e.g. a speed limit sign),
+    this could be implemented in the following way:
+    * periodically inject a known image, of a specific object(s)
+    * expect that within a given time the system will correctly identify the object(s)
+    
+    Such an approach has the advantage of not requiring any special partitioning of
+    the architecture (e.g. which parts must be implemented as user processes and
+    which parts must be kernel components).
+    
+    However, to work, it must also satisfy several conditions:
+    * the monitoring must happen form a context that is at least of equal ASIL level
+    * the monitoring context must not be affected by the interference it is supposed to detect
+    * the testing must have sufficient coverage for detecting all the relevant types of inteference
+    * the testing must account for any recently collected hystorical data
+      that the system might rely on.
+      (e.g. lighting statistics from past frames might get corrupted and affect
+      the following evaluations, but if they are not part of the periodic self test,
+      this corruption wil lgo unnoticed, because the system tested is different from
+      the system in real use)
+
+    The requirements of the Fault Deteting Time Interval (FTDI) and the time required
+    for performing a self-test will dictate how often this self test must be performed.
+
+    Should it be required to ensure availability, it could be possible to use multiple
+    instances of the detection system, as long as it can be proven that they are
+    sufficiently decoupled to not be victim of the same interference, simultaneously.
```

Comment:
This is not correct. If Linux, in this case the monitored entity, is the only "channel" driving the actuator, you can have as many monitors as you like but you'll never be able to recover from a wrong actuator output.
For this to happen you need a 2oo3 architecture where at least 2 channels are capable of producing redundant actuator output and a monitor is able to tell which of the two channel is broken in case of compare mismatch and hence able to pick the output from the sane channel

Reply:
ok for the numbers, I wrote multiple, but that would have included also 2, which is not sufficient

regarding the actuator, I had in mind exactly what you described, meaning that the acutator receives the vetted output, however I agree that it was not as clear as it coudl have been - I will reword that too

Reply:
done